### ISSUE TITLE
Consider a rule for the OGP `og:image:alt` alternate text property

### ISSUE THREAD

**Issue by Malvoz** [dequelabs/axe-core#3355]
The Open Graph Protocol is the de-facto standard for authors to describe resources in regards to sharing. Consider warning developers if the [`og:image`](https://ogp.me/#:~:text=og:image) property is used without [`og:image:alt`](https://ogp.me/#:~:text=og:image:alt).

Such a rule is essentially [Images must have alternate text](https://dequeuniversity.com/rules/axe/4.3/image-alt?application=RuleDescription), but for images that are intended to be shared on other sites.

**Comment by WilcoFiers** [comment]
I'm not too familiar with OGP. Just giving this a quick read:

> og:image:alt - A description of what is in the image (not a caption). If the page specifies an og:image it should specify og:image:alt.

It doesn't sound like a hard requirement to me. Can you elaborate on how this would be used?

**Comment by Malvoz** [comment]
> It doesn't sound like a hard requirement to me.

The end result without `og:image:alt` would in theory be the same as `<img>` without `alt` once a user has shared a URL on a social site or email, so I think it's a valid concern.


For example, if I share https://www.deque.com/axe/ in an email, the client may produce the following:

![axe-share](https://user-images.githubusercontent.com/26493779/149790723-13e426ba-6697-47a2-9dd3-5c3c7c848ddb.png)

as it pulls in the `<meta property="og:...">` properties from that document:

```html
<meta property="og:locale" content="en_US" />
<meta property="og:type" content="article" />
<meta property="og:title" content="axe: Accessibility Testing Tools and Software" />
<meta property="og:description" content="axe is the world’s most popular accessibility testing toolkit built. Visit our website to start building accessibility into your development process." />
<meta property="og:url" content="https://www.deque.com/axe/" />
<meta property="og:site_name" content="Deque" />
<meta property="article:publisher" content="https://www.facebook.com/dequesystems/" />
<meta property="article:modified_time" content="2021-10-19T17:11:27+00:00" />
<meta property="og:image" content="https://www.deque.com/wp-content/uploads/2018/03/DQ_SecondaryLogo_HeroBlue_RGB-1.png" />
<meta property="og:image:width" content="921" />
<meta property="og:image:height" content="921" />
```

(note that this document happens to not have `og:image:alt` specified)

However, it seems that both Twitter and Facebook ignore the property: https://www.stefanjudis.com/today-i-learned/how-to-define-open-graph-twitter-image-alt-text-and-why-it-might-not-matter/. What a shame. 😞

Feel free to close this issue.

**Comment by WilcoFiers** [comment]
@Malvoz Thanks for sharing. I think it's a solid idea. My biggest hesitation wouldn't even be that FB / Twitter aren't using it, but that these images might be decorative and so wouldn't need an alt.
